### PR TITLE
Fixed #9104 -- Moved FieldDoesNotExist to core.exceptions

### PR DESCRIPTION
**Pull request by PirosB3**
Cherry pick of f96990ff71d30a7397b07367e93404d816ad085b

### CODE REVIEW

File: django/db/models/fields/__init__.py [hunk 1]
```diff
@@ -61,10 +60,6 @@ def _load_field(app_label, model_name, field_name):
     return apps.get_model(app_label, model_name)._meta.get_field_by_name(field_name)[0]
 
 
-class FieldDoesNotExist(Exception):
```
Comment (timgraham) [inline django/db/models/fields/__init__.py]: We need to keep an alias for backwards compatibility.

File: django/core/exceptions.py [hunk 2]
```diff
@@ -5,6 +5,11 @@
 from django.utils.encoding import force_text
 
 
+class FieldDoesNotExist(Exception):
+    """The requested field does not exist"""
```
Comment (timgraham) [inline django/core/exceptions.py]: clarify: "model field"

**Comment by timgraham** [comment]
Could you please document the exception in `docs/ref/exceptions.txt` and use "Fixed #9104 -- Moved FieldDoesNotExist to core.exceptions" as the commit message.

File: django/db/models/fields/__init__.py [hunk 3]
```diff
@@ -37,11 +37,10 @@
     'AutoField', 'BLANK_CHOICE_DASH', 'BigIntegerField', 'BinaryField',
     'BooleanField', 'CharField', 'CommaSeparatedIntegerField', 'DateField',
     'DateTimeField', 'DecimalField', 'DurationField', 'EmailField', 'Empty',
-    'Field', 'FieldDoesNotExist', 'FilePathField', 'FloatField',
```
Comment (timgraham) [inline django/db/models/fields/__init__.py]: likewise, should keep it here

File: django/core/exceptions.py (hunk 2 above)
Comment (claudep) [inline django/core/exceptions.py]: I'd move the exception above or below ObjectDoesNotExist

File: django/db/models/query_utils.py [hunk 4]
```diff
@@ -99,7 +99,7 @@ def __get__(self, instance, owner):
         Retrieves and caches the value from the datastore on the first lookup.
         Returns the cached value.
         """
-        from django.db.models.fields import FieldDoesNotExist
+        from django.core.exceptions import FieldDoesNotExist
```
Comment (claudep) [inline django/db/models/query_utils.py]: The import can be moved to the top.

File: django/db/models/fields/__init__.py (hunk 1 above)
Comment (PirosB3) [inline django/db/models/fields/__init__.py]: Done

**Comment by timgraham** [comment]
merged in 8958170755b37ce346ae5257c1000bd936faa3b0, thanks. You can go ahead and make `get_field()` a link in the other PR.

**Pull request closed without merging**